File: SetMismatch.go
func findErrorNums(nums []int) []int {
  n := len(nums)
  mp := map[int]int{}

  for _, num := range nums {
    mp[num]++
  }

  var duplicate, missing int
  for num, count := range mp {
    if count == 2 {
      duplicate = num
    } else if count == 0 {
      missing = num
    }
  }

  if missing == 0 {
    for i := 1; i <= n; i++ {
      if _, ok := mp[i]; !ok {
        missing = i
        break
      }
    }
  }

  return []int{duplicate, missing}
}

File: SetMismatch.cpp
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        int n = nums.size();

        unordered_map<int, int> map;
        for(int i = 1; i <= n; i++) map[i]++;

        for(auto i: nums) map[i]--;

        int duplicate = 0, missing = 0;

        for(auto i: map) {
            if(i.second == -1) duplicate = i.first;
            if(i.second == 1) missing = i.first;
        }

        return {duplicate, missing};
    }
};